utonics/labproject/main.go

```diff
@@ -272,19 +400,111 @@ func getAvailableOrgsAndTeams(botClient, userClient *worker.Client) (map[string]
 	return validOrgTeams, nil
 }
 
-func readPassfile(filename string) (string, string) {
-	passfile, err := os.Open(filename)
+func readConfig(filename string) *labProjectConfig {
+	confFile, err := os.Open(filename)
 	if err != nil {
 		log.Fatal(err)
 	}
-	passdata, err := ioutil.ReadAll(passfile)
+	confData, err := ioutil.ReadAll(confFile)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	userpass := make(map[string]string)
-	if err := json.Unmarshal(passdata, &userpass); err != nil {
+	config := new(labProjectConfig)
+	if err := json.Unmarshal(confData, config); err != nil {
 		log.Fatal(err)
 	}
-	return userpass["username"], userpass["password"]
+
+	if config.Config == nil {
+		config.Config = new(tonic.Config)
+	}
+
+	// Set defaults for any unset values
+	if config.GIN.Web == "" {
+		config.GIN.Web = "https://gin.dev.g-node.org:443"
```

Comment:
Should we add a ToDo here to set an appropriate default value after the development stage?

Reply:
Hm.  You know, it's probably best to make it fail if it's not set.  The address the service talks to is important enough to not rely on an implied default.  This service isn't expected to run against the main GIN service, so we wouldn't want to add that one like we would with something like Valid or DOI.

Let me update this.